image-processing/main.py

```diff
@@ -20,10 +23,43 @@ def trigger_event(event, context):
         None; the output is written to Stackdriver Logging
     """
 
-    print("Event ID: {}".format(context.event_id))
-    print("Event type: {}".format(context.event_type))
-    print("Bucket: {}".format(event["bucket"]))
-    print("File: {}".format(event["name"]))
-    print("Metageneration: {}".format(event["metageneration"]))
-    print("Created: {}".format(event["timeCreated"]))
-    print("Updated: {}".format(event["updated"]))
+    bucket_name = event["bucket"]
+    source_img = event["name"]
+    tmp_download_path = f"/tmp/{source_img}"
+
+    try:
+        download_image(bucket_name, source_img, tmp_download_path)
```

Comment:
So the logic here is when some event is triggered, then the method calls download_image?

Reply:
yep trigger_event is called when an image is uploaded

Reply:
So when images are uploaded then this trigger is called, and the download starts after that.

Reply:
yes user uploads image -> firebase puts it in cloud storage -> this function is triggered and runs all logic inside it